src/main/java/ui/GUIController.java

```diff
@@ -0,0 +1,297 @@
+package ui;
+
+import backend.interfaces.IModel;
+import backend.resource.TurboIssue;
+import filter.expression.FilterExpression;
+import filter.expression.Qualifier;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import javafx.collections.transformation.SortedList;
+import javafx.collections.transformation.TransformationList;
+import ui.issuecolumn.ColumnControl;
+import ui.issuecolumn.IssueColumn;
+import ui.issuecolumn.UIBrowserBridge;
+import util.events.ModelUpdatedEvent;
+import util.events.ModelUpdatedEventHandler;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Comparator;
+import java.util.ArrayList;
+import java.util.function.Predicate;
+
+/**
+ * This class manages the state of UI components and acts as a gateway between back-end components and
+ * GUI components. Any mutation of GUI components should be carried out here.
+ */
+
+public class GUIController {
+    private ColumnControl columnControl;
+    private UI ui;
+
+    // Synchronised with Logic's latest updated multimodel.
+    // Not to be modified from this point or any further in the GUI.
+    private IModel multiModel;
+
+    public GUIController(UI ui, ColumnControl columnControl) {
+        this.ui = ui;
+        this.columnControl = columnControl;
+
+        // Set up the connection to the browser
+        new UIBrowserBridge(ui);
+
+        // Then register model update event handler
+        registerEvents();
+    }
+
+    public void registerEvents() {
+        UI.events.registerEvent((ModelUpdatedEventHandler) this::modelUpdated);
+    }
+
+    /**
+     * The handler method for a ModelUpdatedEvent. It extracts the issues of the multimodel carried by the event,
+     * and then filters and sorts the issues to place each panel in ColumnControl based on these issues.
+     *
+     * The filtering process also produces a list of issues to request metadata for, if e.hasMetadata is false
+     * and their respective panels specify the display of metadata (through the UPDATED filter).
+     *
+     * Based on the hasMetadata state of the model, as specified by the event, the issues will then be displayed
+     * on their respective panels, or held back until their metadata requests have been fired and the downloaded
+     * metadata come back as a subsequent ModelUpdatedEvent (with e.hasMetadata being true).
```

Comment:
I think the problem here is that `updated` can be interpreted either as a filter (which should happen quickly) or a functional keyword (which triggers a metadata update, for which it is acceptable to tell the user to wait).

If `updated` is interpreted as a filter, then it makes more sense to give the user immediate visual feedback, but if it's interpreted as a command ("download metadata for me") then as a user, I think it's OK to wait for a while.

Though, since it's a filter box, it's better to interpret it primarily as a filter keyword. I'll make a separate issue for this.